### PR TITLE
Fancy mesh rendering examples

### PR DESCRIPTION
**Pull request by AnkeAnke**
Adding minimum alpha setting for Martin.

Adding example workspaces for mesh rasterization:

Different alpha settings (opaque, uniform, angle-based, density-based/smoke surface)
exAlphas
 
Geometry contouring (edges, silhouettes, illustration and hatching)
exContours

Multiple mesh rendering (compositing with other renderers, transforming rasterizations)
exCow

Composite renderings (as for paper renderings)
exComposites

### CODE REVIEW

File: modules/meshrenderinggl/glsl/fancymeshrenderer.frag [hunk 1]
```diff
@@ -187,27 +188,31 @@ vec4 performShading() {
...
 #ifdef ALPHA_ANGLE_BASED
         alpha *= pow(acos(angle) * 2 / M_PI, alphaSettings.angleExp);
+        alpha += alphaSettings.minAlpha;
 #endif
 #ifdef ALPHA_NORMAL_VARIATION
         float nv_dzi = dFdxFinest(normal.z);
         float nv_dzj = dFdyFinest(normal.z);
         float nv_curvature = min(1, nv_dzi * nv_dzi + nv_dzj * nv_dzj + 0.0000001);
         alpha *= min(1, pow(nv_curvature, alphaSettings.normalExp * 0.5) * 10);
+        alpha += alphaSettings.minAlpha;
```
Comment (petersteneteg) [inline modules/meshrenderinggl/glsl/fancymeshrenderer.frag:202]: will this not give a max of 1+minAlpha? there are some off constants in this code :)

File: modules/meshrenderinggl/src/rendering/fragmentlistrenderer.cpp [hunk 2]
```diff
@@ -146,9 +146,6 @@ bool FragmentListRenderer::postPass(bool useIllustration, const Image* backgroun
     // check if enough space was available
     if (numFrags > fragmentSize_) {
         // we have to resize the fragment storage buffer
-        LogInfo("fragment lists resolved, pixels drawn: "
```
Comment (petersteneteg) [inline modules/meshrenderinggl/src/rendering/fragmentlistrenderer.cpp:149]: maybe we could expose the memory usage as a readonly property or something? other PR probably...

**Review by petersteneteg — CHANGES_REQUESTED** [review]

File: modules/meshrenderinggl/src/rendering/fragmentlistrenderer.cpp (hunk 2 above)
Comment (AnkeAnke) [reply modules/meshrenderinggl/src/rendering/fragmentlistrenderer.cpp:149]: Maybe. On the other hand I never experienced any memory overflow on a 4K screen, so hopefully the error will never be a problem.

File: modules/meshrenderinggl/glsl/fancymeshrenderer.frag (hunk 1 above)
Comment (AnkeAnke) [reply modules/meshrenderinggl/glsl/fancymeshrenderer.frag:202]: Very true.

File: modules/meshrenderinggl/src/rendering/fragmentlistrenderer.cpp (hunk 2 above)
Comment (petersteneteg) [reply modules/meshrenderinggl/src/rendering/fragmentlistrenderer.cpp:149]: When i rewrote that code I did some tests, and as long as you only use one, renderer it not to bad. but If you have a couple of renderers and use illustration buffers, then you easily use several GBs of memory...

**Review by petersteneteg — APPROVED** [review]

**Comment by martinfalk** [comment]
Nice!

**Pull request merged**